### PR TITLE
Add OpInfos for torch.{complex, view_as_real, view_as_complex}

### PR DESCRIPTION
**Pull request by anjali411**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56524 Add OpInfos for torch.{complex, view_as_real, view_as_complex}**

Differential Revision: [D27909165](https://our.internmc.facebook.com/intern/diff/D27909165)

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -5674,14 +5698,8 @@ def unpack_variables(args):
 
 
 EXCLUDE_FUNCTIONAL = {
-    'addmm',
```
Comment (anjali411) [inline torch/testing/_internal/common_methods_invocations.py:5677]: cleanup

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -2493,13 +2503,13 @@ def sample_inputs_matmul(op_info, device, dtype, requires_grad):
...
-def sample_inputs_polar(op_info, device, dtype, requires_grad, **kwargs):
-    def _make_tensor_helper(shape, low=None, high=None):
-        return make_tensor(shape, device, dtype, low=low, high=high, requires_grad=requires_grad)
+def sample_inputs_complex_polar(op_info, device, dtype, requires_grad, **kwargs):
+    def _make_tensor_helper(shape):
+        return make_tensor(shape, device, dtype, requires_grad=requires_grad)
 
     samples = (
-        SampleInput(_make_tensor_helper((S, S), low=0), args=(_make_tensor_helper((S, S)),)),
-        SampleInput(_make_tensor_helper((), low=0), args=(_make_tensor_helper(()),)),
```
Comment (anjali411) [inline torch/testing/_internal/common_methods_invocations.py:2502]: TODO: undo this change since abs (first input to polar) should be non-negative

File: test/test_autograd.py [hunk 3]
```diff
@@ -5368,7 +5368,7 @@ def run_functional_checks(test_case, test_name, name, apply_fn, run_grad_checks,
 # the tests for these ops which do not have 'complex' in variant should not run for complex
 # and only run for floating point
 
-separate_complex_tests = ['view_as_real', 'div', '__rdiv__', 'sub']
+separate_complex_tests = ['div', '__rdiv__', 'sub']
```
Comment (mruberry) [inline test/test_autograd.py:5371]: Wow; we're almost done with this list! That's cool

**Review by mruberry — COMMENTED** [review]
Awesome!

Some failures are real, however. 

Lint: 

```
./torch/testing/_internal/common_methods_invocations.py:3388:17: E126 continuation line over-indented for hanging indent
```

the new polar grad and gradgrad tests are failing, and a skip is needed for:

test_supported_backward_view_as_complex_cuda_float16

The skip and lint fix should be easy. For polar grad and gradgrad this could revert modifying its sample inputs and possible file an issue?

**Review by mruberry — APPROVED** [review]
Awesome!
